app/models/db_connection.rb

```diff
@@ -47,58 +62,35 @@ def password(key)
   #     initialization_vector: <Base64 enctoded string>,
   #     key: string
   #   }
-  def self.encrypt(key:, password:, iv: nil)
+  def self.encrypt(key:, password:, initialization_vector: nil)
     error!('No AES key sent', 401) unless key
 
     aes_key = Base64.strict_decode64(key)
     cipher = OpenSSL::Cipher::AES.new(256, :CBC)
-    iv = iv ? Base64.strict_decode64(iv) : cipher.random_iv
-    cipher.iv = iv
+    initialization_vector = if initialization_vector.nil?
+                              cipher.random_iv
+                            else
+                              Base64.strict_decode64(initialization_vector)
+                            end
+    cipher.iv = initialization_vector
```

Comment:
What about adding code that fails if an IV is provided production mode? It would make sure that you can't accidentally reuse an IV.

Reply:
Good point, thanks :+1: 
The iv param was used only to make the code easyier testable. Since factory bot got refactored in [this PR](https://github.com/lebalz/db-sql/pull/7) and an iv is always generated by random, this should then not be a problem anymore.